pkg/asset/machines/openstack/machines.go

```diff
@@ -56,31 +56,33 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 		total = *pool.Replicas
 	}
 	machines := make([]machineapi.Machine, 0, total)
-	providerConfigs := map[string]*machinev1alpha1.OpenstackProviderSpec{}
```

Comment:
This map was used to accumulate identical providerSpecs so that they could be reused. This optimisation is now problematic because we distribute differently; I have removed it.